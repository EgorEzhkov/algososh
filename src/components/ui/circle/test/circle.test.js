import renderer from "react-test-renderer";
import { ElementStates } from "../../../../types/element-states";
import { Button } from "../../button/button";

import { Circle } from "../circle";

describe("Тестирование компонента Circle", () => {
  it("Circle правильно отображается без буквы", () => {
    const circle = renderer.create(<Circle></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается c буквами", () => {
    const circle = renderer.create(<Circle letter="asdf"></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается c head", () => {
    const circle = renderer.create(<Circle head="asdf"></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается c react-элементом в head", () => {
    const circle = renderer.create(<Circle head={<Button />}></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается c tail", () => {
    const circle = renderer.create(<Circle tail="asdf"></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается c react-элементом в tail", () => {
    const circle = renderer.create(<Circle tail={<Button />}></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается c пропом isSmall ===  true", () => {
    const circle = renderer.create(<Circle isSmall={true}></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается в состоянии changing", () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing}></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается в состоянии default", () => {
    const circle = renderer.create(<Circle state={ElementStates.Default}></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it("Circle правильно отображается в состоянии modified", () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified}></Circle>).toJSON();
    expect(circle).toMatchSnapshot();
  });
});
