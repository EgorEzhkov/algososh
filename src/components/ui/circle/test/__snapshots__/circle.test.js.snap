// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle Circle правильно отображается c head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    asdf
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается c react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <button
      className="text text_type_button text_color_primary button undefined false "
      type="button"
    >
      <p
        className="text undefined"
      />
    </button>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается c react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <button
      className="text text_type_button text_color_primary button undefined false "
      type="button"
    >
      <p
        className="text undefined"
      />
    </button>
  </div>
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается c tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    asdf
  </div>
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается c буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      asdf
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается c пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle правильно отображается в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
